From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: UN1CA y2slte <un1ca-y2slte@localhost>
Date: Sat, 30 Aug 2026 00:00:00 -0300
Subject: [PATCH] Allow custom booster thresholds values

Reload each Android 16 product-feature literal immediately before parsing it.
---
 .../samsung/android/server/wifi/SemFrameworkFacade.smali | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/smali/com/samsung/android/server/wifi/SemFrameworkFacade.smali b/smali/com/samsung/android/server/wifi/SemFrameworkFacade.smali
--- a/smali/com/samsung/android/server/wifi/SemFrameworkFacade.smali
+++ b/smali/com/samsung/android/server/wifi/SemFrameworkFacade.smali
@@ -314,6 +314,6 @@
 .method public getBoosterThresholds()[I
     .locals 3
 
-    const-string p0, "0"
+    const-string p0, "CONFIG_DATA_ACTIVITY_AFFINITY_BOOSTER_THRESHOLD"
 
     invoke-static {p0}, Ljava/lang/Integer;->parseInt(Ljava/lang/String;)I
@@ -326,6 +326,8 @@
 
     div-int/lit8 v0, v0, 0x8
 
+    const-string p0, "CONFIG_CPU_CSTATE_DISABLE_THRESHOLD"
+
     invoke-static {p0}, Ljava/lang/Integer;->parseInt(Ljava/lang/String;)I
 
     move-result v2
@@ -334,4 +336,6 @@
 
     div-int/lit8 v2, v2, 0x8
 
+    const-string p0, "CONFIG_L1SS_DISABLE_THRESHOLD"
+
     invoke-static {p0}, Ljava/lang/Integer;->parseInt(Ljava/lang/String;)I
-- 
2.51.0
